components/GradientBox.vue

```diff
@@ -0,0 +1,50 @@
+<script>
+export default {
+  props: {
+    primaryColorVar: {
+      type:     String,
+      required: true
+    },
+    showTab: {
+      type:     Boolean,
+      default: true,
+    }
+  },
+
+  computed: {
+    leftColor() {
+      return this.customizePrimaryColorOpacity(0.25);
+    },
+    rightColor() {
+      return this.customizePrimaryColorOpacity(0.125);
+    },
+    primaryColor() {
+      return this.customizePrimaryColorOpacity(1);
+    },
+    style() {
+      const background = `background: transparent linear-gradient(94deg, ${ this.leftColor } 0%, ${ this.rightColor } 100%) 0% 0% no-repeat padding-box;`;
+      const border = this.showTab ? `border-left: 9px solid ${ this.primaryColor };` : '';
+
+      return `${ background }${ border }`;
```

Comment:
I'm just going to merge this it for now for the demo but the literal gradient should just be the CSS variable's value.  JS generating it is more complicated and less flexible, and it's likely that the light theme will want a different direction, stops, opacity etc that this won't allow.

See https://github.com/rancher/dashboard/pull/877/files#diff-e2aaa9e67fbaa73921bcd1cd22f06b53R175-R193 for example.